botocore/crt/auth.py

```diff
@@ -494,6 +494,10 @@ def _apply_signing_changes(self, aws_request, signed_crt_request):
         # fragment - 4
         aws_request.url = urlunsplit((p[0], p[1], p[2], signed_query, p[4]))
 
+    def _should_add_content_sha256_header(self, existing_sha256, explicit_payload):
+        # only add X-Amz-Content-SHA256 header if header already set to UNSIGNED_PAYLOAD
+        return existing_sha256 == UNSIGNED_PAYLOAD
```

Comment:
Why are we only adding this if it's unsigned? What if we have an actual payload we're signing for?

Reply:
It's a temporary workaround we want to add for vpc-lattice and not want to introduce the change for others (Even though that it might be more reasonable to add the header to sign for other service as well, and it will match the behavior for the botocore signer).

We decided to have an other workaround as the short term patch within CRT now. https://github.com/awslabs/aws-c-auth/pull/249

We will have further discussion about what's the proper way to handle their use case in the near future, and it should be across SDKs not only for CRT.

So, no need for this PR now.